carapace-server/src/test/java/org/carapaceproxy/cluster/impl/ZooKeeperGroupMembershipHandlerTest.java

```diff
@@ -107,17 +109,14 @@ public void reconnected() {
 
                 peer1.fireEvent("foo");
 
-                for (int i = 0; i < 10; i++) {
-                    if (eventFired2.get() >= 1) {
-                        break;
-                    }
-                    Thread.sleep(100);
-                }
+                TestUtils.waitForCondition(() -> {
+                    return eventFired2.get() >= 1;
+                }, 100);
                 assertTrue(eventFired2.get() >= 1);
```

Comment:
waitForCondition will fail if the condition is not met in time so this assert is redundant